catalog/src/main/java/org/killbill/billing/catalog/DefaultVersionedCatalog.java

```diff
@@ -190,6 +190,21 @@ private void validatePlanShape(final Plan plan, final Plan targetPlan, final Val
                                                              cur.getName(), plan.getName(), plan.getCatalog().getEffectiveDate(), targetPlan.getCatalog().getEffectiveDate()),
                                                DefaultVersionedCatalog.class, ""));
             }
+	    //Fix for https://github.com/killbill/killbill/issues/1465
+            validatePlanDuration(plan, cur, errors);
+        }
+    }
+    
+    private void validatePlanDuration(final Plan plan, final PlanPhase cur, final ValidationErrors errors) {
+        if(cur.getPhaseType().name().equals(PhaseType.EVERGREEN.name()) && !cur.getDuration().getUnit().name().equals(TimeUnit.UNLIMITED.name())) {
+            errors.add(new ValidationError(String.format("EVERGREEN Phase '%s'for plan '%s' in version '%s'  must have duration as UNLIMITED'",
```

Comment:
Missing space before `for plan`.
Double spaces (extra space) before `must`.

Reply:
Fixed.